File: 10799/a.cpp
#include<cstdio>
#include<stack>
using namespace std;
int main(){
  stack<char> s;
  char arr[100000] = {};
  scanf("%s", arr);
  int cnt = 0;
  for(int i=0; i<10000; i++){
    if(arr[i] == '\0') break;
    cnt++;
  }
  for(int i=0; i<cnt; i++){
    s.push(arr[i]);
    
  }
}

File: 10799/a.py
x = input()
stack = []
cnt = 0
for i in range(len(x)):
    if x[i] == '(':
        stack.append(x[i])
    else:
        stack.pop()
        if(x[i-1] == '('):
            cnt += len(stack)
        else:
            cnt += 1

print(cnt)
